File: .tmp_py4013/batch_08_dump/2892_minimizing_array_after_replacing_pairs_with_their_product.js
// LeetCode 2892 - Minimizing Array After Replacing Pairs With Their Product
// https://leetcode.com/problems/minimizing-array-after-replacing-pairs-with-their-product/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minArrayLength = function(nums, k) {
    if (nums.length === 0) return 0;
    let ans = 1;
    let prod = nums[0];
    for (let i = 1; i < nums.length; i++) {
        if (prod <= k && nums[i] <= k && (nums[i] === 0 || prod <= Math.floor(k / nums[i]))) {
            prod *= nums[i];
        } else {
            ans++;
            prod = nums[i];
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check_12/2892_minimizing_array_after_replacing_pairs_with_their_product.rs
struct Solution;
// LeetCode 2892 - Minimizing Array After Replacing Pairs With Their Product
// https://leetcode.com/problems/minimizing-array-after-replacing-pairs-with-their-product/

impl Solution {
    pub fn min_array_length(nums: Vec<i32>, k: i32) -> i32 {
        if nums.is_empty() {
            return 0;
        }
        let mut ans = 1;
        let mut prod = nums[0] as i64;
        let k = k as i64;
        for &v in nums.iter().skip(1) {
            let v = v as i64;
            if prod <= k && v <= k && (v == 0 || prod <= k / v) {
                prod *= v;
            } else {
                ans += 1;
                prod = v;
            }
        }
        ans
    }
}

fn main() {}
